### PR TITLE
translation db and asynchronous listener

### PR DESCRIPTION
**Pull request by danielzhang130**
- Add program and comment translation usage to database
- Add cron job to keep Translation database tables small
- Behat tests

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

### CODE REVIEW

**Comment by danielzhang130** [comment]
Main differences compared to 
https://github.com/danielzhang130/Catroweb/pull/31

1. Add new column 'usage_per_month'.
This value in theory can be calculated by the admin dashboard from other values in the db. But it turns out to be very hard to add sorting to the admin table this way. The admin table seems only to sort using sql and not just an inplace sort of the html content.

2. Add ability to trim storage for different date limit for programs and comments independently. 
This change combined with the ability to trigger this command from the admin dashboard should address the potential issue that comments table grows too fast.

**Comment by danielzhang130** [comment]
I took this approach for the comments table (deviated from what we planned in the wiki) mainly because we could be all well predicting what's going to happen, but I thought you never know for sure if you don't log enough data. So, to start, let's log a bit more.

File: src/Entity/Translation/MachineTranslation.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+   */
+  public function preUpdate(): void
+  {
+    $this->last_modified_at = TimeUtils::getDateTime();
+
+    $date_interval = $this->last_modified_at->diff($this->created_at);
+    $months = $date_interval->days / 30;
+
+    if (1 > $months) {
+      $this->usage_per_month = $this->usage_count;
```
Comment (JoshuaSchaefer) [inline src/Entity/Translation/MachineTranslation.php:132]: Will this cause any problems by setting a float to an int without some sort of casting?

File: src/Entity/Translation/MachineTranslation.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    } else {
+      $this->usage_per_month = $this->usage_count / $months;
```
Comment (JoshuaSchaefer) [inline src/Entity/Translation/MachineTranslation.php:134]: Same applies here, will there be any type issues with this?

Comment (danielzhang130) [reply src/Entity/Translation/MachineTranslation.php:134]: no. it still shows decimal values

File: src/Entity/Translation/MachineTranslation.php (hunk 1 above)
Comment (danielzhang130) [reply src/Entity/Translation/MachineTranslation.php:132]: not that i can tell

**Comment by JoshuaSchaefer** [comment]
Everything looks good to me, great job. I really like the new commands to be able to prune the project and comment tables separately since I could definitely see the comment table getting too big before the project table does. I also agree with logging as much data as we can at the beginning to see how it goes and then we can always modify or reduce the amount later if it becomes too much.

**Pull request closed without merging**